In [30]:
%load_ext sql

The sql extension is already loaded. To reload it, use:
  %reload_ext sql


In [31]:
%sql mysql://USERNAME:PASSWORD@HOST/DATABASE

1.My primary question is to find out who is the most statistically efficient basketball player is. Because of this I was curious to find out which players made over 10 field goals through out the course of the season. This will help me better understand and keep track of the elite players and their player id's.

In [32]:
%%sql
SELECT DISTINCT Player_Id,
Player_name,
Team_city,
FGM
FROM game_details
WHERE FGM > 10;

 * mysql://admin:***@sqlproject.cumiuvaqlsmj.us-east-1.rds.amazonaws.com/sports_data
148 rows affected.


[(1628389, 'Bam Adebayo', 'Miami', 11),
 (1627742, 'Brandon Ingram', 'New Orleans', 11),
 (203999, 'Nikola Jokic', 'Denver', 13),
 (1627759, 'Jaylen Brown', 'Boston', 12),
 (1629012, 'Collin Sexton', 'Cleveland', 13),
 (1626164, 'Devin Booker', 'Phoenix', 11),
 (1629029, 'Luka Doncic', 'Dallas', 13),
 (1628381, 'John Collins', 'Atlanta', 11),
 (202699, 'Tobias Harris', 'Philadelphia', 12),
 (1629022, 'Lonnie Walker IV', 'San Antonio', 13),
 (1629012, 'Collin Sexton', 'Cleveland', 12),
 (1628368, "De'Aaron Fox", 'Sacramento', 11),
 (202681, 'Kyrie Irving', 'Brooklyn', 19),
 (203932, 'Aaron Gordon', 'Orlando', 14),
 (203952, 'Andrew Wiggins', 'Golden State', 14),
 (203999, 'Nikola Jokic', 'Denver', 14),
 (1627750, 'Jamal Murray', 'Denver', 12),
 (1629029, 'Luka Doncic', 'Dallas', 15),
 (203468, 'CJ McCollum', 'Portland', 11),
 (204060, 'Joe Ingles', 'Utah', 12),
 (1628378, 'Donovan Mitchell', 'Utah', 16),
 (203078, 'Bradley Beal', 'Washington', 16),
 (201566, 'Russell Westbrook', 'Washington', 12),
 (1626157, 'Karl-Anthony Towns', 'Minnesota', 15),
 (1630162, 'Anthony Edwards', 'Minnesota', 15),
 (1626164, 'Devin Booker', 'Phoenix', 13),
 (203081, 'Damian Lillard', 'Portland', 12),
 (2544, 'LeBron James', 'Los Angeles', 14),
 (1626181, 'Norman Powell', 'Toronto', 14),
 (201935, 'James Harden', 'Brooklyn', 13),
 (203507, 'Giannis Antetokounmpo', 'Milwaukee', 13),
 (203078, 'Bradley Beal', 'Washington', 12),
 (1628369, 'Jayson Tatum', 'Boston', 11),
 (1627759, 'Jaylen Brown', 'Boston', 11),
 (203897, 'Zach LaVine', 'Chicago', 12),
 (1629008, 'Michael Porter Jr.', 'Denver', 11),
 (1629029, 'Luka Doncic', 'Dallas', 16),
 (1628369, 'Jayson Tatum', 'Boston', 12),
 (202710, 'Jimmy Butler', 'Miami', 11),
 (202699, 'Tobias Harris', 'Philadelphia', 11),
 (203897, 'Zach LaVine', 'Chicago', 15),
 (203506, 'Victor Oladipo', 'Houston', 13),
 (1629627, 'Zion Williamson', 'New Orleans', 11),
 (203081, 'Damian Lillard', 'Portland', 13),
 (1630162, 'Anthony Edwards', 'Minnesota', 11),
 (1626149, 'Montrezl Harrell', 'Los Angeles', 11),
 (203507, 'Giannis Antetokounmpo', 'Milwaukee', 12),
 (203078, 'Bradley Beal', 'Washington', 13),
 (203944, 'Julius Randle', 'New York', 13),
 (202681, 'Kyrie Irving', 'Brooklyn', 13),
 (203999, 'Nikola Jokic', 'Denver', 11),
 (202685, 'Jonas Valanciunas', 'Memphis', 11),
 (203952, 'Andrew Wiggins', 'Golden State', 12),
 (202696, 'Nikola Vucevic', 'Orlando', 15),
 (1630162, 'Anthony Edwards', 'Minnesota', 12),
 (1626181, 'Norman Powell', 'Toronto', 13),
 (1629627, 'Zion Williamson', 'New Orleans', 13),
 (1629628, 'RJ Barrett', 'New York', 12),
 (1629060, 'Rui Hachimura', 'Washington', 11),
 (201566, 'Russell Westbrook', 'Washington', 16),
 (1626157, 'Karl-Anthony Towns', 'Minnesota', 12),
 (203076, 'Anthony Davis', 'Los Angeles', 13),
 (201144, 'Mike Conley', 'Utah', 11),
 (203081, 'Damian Lillard', 'Portland', 11),
 (1627759, 'Jaylen Brown', 'Boston', 15),
 (1628381, 'John Collins', 'Atlanta', 12),
 (201142, 'Kevin Durant', 'Brooklyn', 11),
 (2544, 'LeBron James', 'Los Angeles', 11),
 (1627749, 'Dejounte Murray', 'San Antonio', 12),
 (202696, 'Nikola Vucevic', 'Orlando', 12),
 (1629008, 'Michael Porter Jr.', 'Denver', 12),
 (1628367, 'Josh Jackson', 'Detroit', 11),
 (1627747, 'Caris LeVert', 'Brooklyn', 12),
 (202330, 'Gordon Hayward', 'Charlotte', 12),
 (1628365, 'Markelle Fultz', 'Orlando', 11),
 (203083, 'Andre Drummond', 'Cleveland', 11),
 (203901, 'Elfrid Payton', 'New York', 12),
 (1627763, 'Malcolm Brogdon', 'Indiana', 11),
 (201939, 'Stephen Curry', 'Golden State', 11),
 (203897, 'Zach LaVine', 'Chicago', 11),
 (1629012, 'Collin Sexton', 'Cleveland', 15),
 (203078, 'Bradley Beal', 'Washington', 14),
 (203944, 'Julius Randle', 'New York', 11),
 (1626174, 'Christian Wood', 'Houston', 14),
 (201935, 'James Harden', 'Houston', 12),
 (203468, 'CJ McCollum', 'Portland', 17),
 (202330, 'Gordon Hayward', 'Charlotte', 11),
 (1626179, 'Terry Rozier', 'Charlotte', 15),
 (1629628, 'RJ Barrett', 'New York', 11),
 (16277

I learned that in my list of 1008 players, 76 of them made over 10 field goals in at least one point of the season. If you are to take a look at the list it is perdominatly made of allstars or at least very talented players.

2.In this next exploratory query I want to look at some of the games in which the home team had the highest overall shooting percentage. This will give me a better insight at the which teams to look into because one of the most efficient players is likley to have a high shooting percentage. 

In [14]:
%%sql
SELECT DISTINCT games.Game_id,
games.Home_team_id,
games.PTS_home,
game_details.Team_city
FROM games
JOIN game_details
ON games.Game_id = game_details.Game_id 
WHERE PTS_home > 130;

 * mysql://admin:***@sqlproject.cumiuvaqlsmj.us-east-1.rds.amazonaws.com/sports_data
26 rows affected.


[(22000006, 1610612749, 138, 'Golden State'),
 (22000006, 1610612749, 138, 'Milwaukee'),
 (22000008, 1610612747, 138, 'Dallas'),
 (22000008, 1610612747, 138, 'Los Angeles'),
 (22000057, 1610612751, 145, 'Atlanta'),
 (22000057, 1610612751, 145, 'Brooklyn'),
 (22000166, 1610612764, 131, 'Utah'),
 (22000166, 1610612764, 131, 'Washington'),
 (22000500, 1610612752, 140, 'Sacramento'),
 (22000500, 1610612752, 140, 'New York'),
 (22000545, 1610612755, 131, 'Utah'),
 (22000545, 1610612755, 131, 'Philadelphia'),
 (22000552, 1610612738, 132, 'Toronto'),
 (22000552, 1610612738, 132, 'Boston'),
 (22000573, 1610612749, 134, 'New York'),
 (22000573, 1610612749, 134, 'Milwaukee'),
 (22000594, 1610612744, 131, 'Utah'),
 (22000594, 1610612744, 131, 'Golden State'),
 (22000595, 1610612755, 134, 'San Antonio'),
 (22000595, 1610612755, 134, 'Philadelphia'),
 (22000601, 1610612740, 135, 'LA'),
 (22000601, 1610612740, 135, 'New Orleans'),
 (22000615, 1610612747, 137, 'Minnesota'),
 (22000615, 1610612747, 137, 'Los Angeles'),
 (22000637, 1610612743, 131, 'Chicago'),
 (22000637, 1610612743, 131, 'Denver')]

As we can see there were 26 times where the home team scored over 130 points. As I take a brief look through this list the teams that have more than one appearance are the ones with candidates I believe could possibly answer my question - the likes of Golden State's Steph Curry, Lebron James on the Lakers, or Donovan Mitchell and Utah.

3.I am now going to try to find players who had a strong fieldgoal percentage while also shooting over a certain amount of times. Going 100% from the floor but only shooting one time will not lead me to find the players I am looking for. 

In [18]:
%%sql
SELECT DISTINCT Player_name,
Team_ab,
Player_id 
FROM game_details
WHERE FG_PCT > 0.6 AND FGM > 8;

 * mysql://admin:***@sqlproject.cumiuvaqlsmj.us-east-1.rds.amazonaws.com/sports_data
91 rows affected.


[('Zion Williamson', 'NOP', 1629627),
 ('Nikola Jokic', 'DEN', 203999),
 ('Rui Hachimura', 'WAS', 1629060),
 ('Montrezl Harrell', 'LAL', 1626149),
 ('Deandre Ayton', 'PHX', 1629028),
 ('Luka Doncic', 'DAL', 1629029),
 ('John Collins', 'ATL', 1628381),
 ('Tobias Harris', 'PHI', 202699),
 ('Lonnie Walker IV', 'SAS', 1629022),
 ('Malcolm Brogdon', 'IND', 1627763),
 ('Kyrie Irving', 'BKN', 202681),
 ('Aaron Gordon', 'ORL', 203932),
 ('Evan Fournier', 'ORL', 203095),
 ('Jordan Poole', 'GSW', 1629673),
 ('Otto Porter Jr.', 'CHI', 203490),
 ('Karl-Anthony Towns', 'MIN', 1626157),
 ('Tim Hardaway Jr.', 'DAL', 203501),
 ('Joe Ingles', 'UTA', 204060),
 ('Bradley Beal', 'WAS', 203078),
 ('Bogdan Bogdanovic', 'ATL', 203992),
 ('LeBron James', 'LAL', 2544),
 ('Norman Powell', 'TOR', 1626181),
 ('Chris Boucher', 'TOR', 1628449),
 ('Jakob Poeltl', 'SAS', 1627751),
 ('Michael Porter Jr.', 'DEN', 1629008),
 ('Jimmy Butler', 'MIA', 202710),
 ('Zach LaVine', 'CHI', 203897),
 ('Danilo Gallinari', 'ATL', 201568),
 ('Victor Oladipo', 'HOU', 203506),
 ('Damian Lillard', 'POR', 203081),
 ('Giannis Antetokounmpo', 'MIL', 203507),
 ('Dejounte Murray', 'SAS', 1627749),
 ('Kristaps Porzingis', 'DAL', 204001),
 ('Jonas Valanciunas', 'MEM', 202685),
 ('Chris Paul', 'PHX', 101108),
 ('Andrew Wiggins', 'GSW', 203952),
 ('Jaylen Brown', 'BOS', 1627759),
 ('Patrick Williams', 'CHI', 1630172),
 ('Kawhi Leonard', 'LAC', 202695),
 ('Clint Capela', 'ATL', 203991),
 ('Domantas Sabonis', 'IND', 1627734),
 ('Doug McDermott', 'IND', 203926),
 ("De'Andre Hunter", 'ATL', 1629631),
 ('Collin Sexton', 'CLE', 1629012),
 ('Thomas Bryant', 'WAS', 1628418),
 ('Richaun Holmes', 'SAC', 1626158),
 ('Christian Wood', 'HOU', 1626174),
 ('Khris Middleton', 'MIL', 203114),
 ('Dwayne Bacon', 'ORL', 1628407),
 ('Nikola Vucevic', 'ORL', 202696),
 ('Isaiah Roby', 'OKC', 1629676),
 ('Monte Morris', 'DEN', 1628420),
 ('Luguentz Dort', 'OKC', 1629652),
 ('James Harden', 'HOU', 201935),
 ('Gary Trent Jr.', 'POR', 1629018),
 ('Andre Drummond', 'CLE', 203083),
 ('Elfrid Payton', 'NYK', 203901),
 ('Darius Garland', 'CLE', 1629636),
 ('Jerami Grant', 'DET', 203924),
 ('Terrence Ross', 'ORL', 203082),
 ('Julius Randle', 'NYK', 203944),
 ('T.J. Warren', 'IND', 203933),
 ('Anthony Davis', 'LAL', 203076),
 ('Gordon Hayward', 'CHA', 202330),
 ('Terry Rozier', 'CHA', 1626179),
 ('RJ Barrett', 'NYK', 1629628),
 ('Victor Oladipo', 'IND', 203506),
 ('Bam Adebayo', 'MIA', 1628389),
 ('Jrue Holiday', 'MIL', 201950),
 ('Trae Young', 'ATL', 1629027),
 ('Ja Morant', 'MEM', 1629630),
 ('Paul George', 'LAC', 202331),
 ('Brandon Ingram', 'NOP', 1627742),
 ('Jaylen Nowell', 'MIN', 1629669),
 ('Josh Richardson', 'DAL', 1626196),
 ('DeMar DeRozan', 'SAS', 201942),
 ('Myles Turner', 'IND', 1626167),
 ('Enes Kanter', 'POR', 202683),
 ('James Harden', 'BKN', 201935),
 ('Buddy Hield', 'SAC', 1627741),
 ('Reggie Jackson', 'LAC', 202704),
 ('Jalen Brunson', 'DAL', 1628973),
 ('Shake Milton', 'PHI', 1629003),
 ('Rudy Gobert', 'UTA', 203497),
 ('Bruce Brown', 'BKN', 1628971),
 ('P.J. Washington', 'CHA', 1629023),
 ('Harrison Barnes', 'SAC', 203084),
 ('Ben Simmons', 'PHI', 1627732),
 ('Marvin Bagley III', 'SAC', 1628963),
 ('Kelly Oubre Jr.', 'GSW', 1626162),
 ('Tyus Jones', 'MEM', 1626145)]

Out of all the players that are in the database, only 91 of them have every played a game where they made over 8 shots shooting at least 60% on the game. In the NBA shooting at 50% is considered a great game, because of this I wanted to find players who exceeded that benchmark making at least 8 shots. 

4.I now want to find out how many players in my dataset have games recorded where they did not actually get in the game. This will help me get a rough understand of which players do not play. (There is some room for interpretation in this finding though because certain players may have gotten injured meaning they have game where they did not play). 

In [21]:
%%sql
SELECT DISTINCT Player_name,
Player_id,
Team_city
FROM game_details
WHERE FGM IS NULL;

 * mysql://admin:***@sqlproject.cumiuvaqlsmj.us-east-1.rds.amazonaws.com/sports_data
306 rows affected.


[('Brian Bowen II', 1628968, 'Indiana'),
 ('Aaron Holiday', 1628988, 'Indiana'),
 ('Kelan Martin', 1629103, 'Indiana'),
 ('JaKarr Sampson', 203960, 'Indiana'),
 ('Cassius Stanley', 1630199, 'Indiana'),
 ('Goran Dragic', 201609, 'Miami'),
 ('Maurice Harkless', 203090, 'Miami'),
 ('KZ Okpala', 1629644, 'Miami'),
 ('Chris Silva', 1629735, 'Miami'),
 ('Max Strus', 1629622, 'Miami'),
 ('Shai Gilgeous-Alexander', 1628983, 'Oklahoma City'),
 ('Darius Miller', 203121, 'Oklahoma City'),
 ('Mike Muscala', 203488, 'Oklahoma City'),
 ('Anthony Lamb', 1630237, 'Houston'),
 ('Ben McLemore', 203463, 'Houston'),
 ('Justin Patton', 1628383, 'Houston'),
 ('Lonzo Ball', 1628366, 'New Orleans'),
 ('Wenyen Gabriel', 1629117, 'New Orleans'),
 ('Willy Hernangomez', 1626195, 'New Orleans'),
 ('Will Magnay', 1630266, 'New Orleans'),
 ('Naji Marshall', 1630230, 'New Orleans'),
 ('Bol Bol', 1629626, 'Denver'),
 ('Vlatko Cancar', 1628427, 'Denver'),
 ('R.J. Hampton', 1630181, 'Denver'),
 ('Markus Howard', 1630210, 'Denver'),
 ('Zeke Nnaji', 1630192, 'Denver'),
 ('Carsen Edwards', 1629035, 'Boston'),
 ('Javonte Green', 1629750, 'Boston'),
 ('Jeff Teague', 201952, 'Boston'),
 ('Tremont Waters', 1629682, 'Boston'),
 ('Troy Brown Jr.', 1628972, 'Washington'),
 ('Anthony Gill', 1630264, 'Washington'),
 ('Jerome Robinson', 1629010, 'Washington'),
 ('Moritz Wagner', 1629021, 'Washington'),
 ('Cassius Winston', 1630216, 'Washington'),
 ('Chris Chiozza', 1629185, 'Brooklyn'),
 ('Timothe Luwawu-Cabarrot', 1627789, 'Brooklyn'),
 ('Reggie Perry', 1629617, 'Brooklyn'),
 ("DeAndre' Bembry", 1627761, 'Toronto'),
 ('Terence Davis', 1629056, 'Toronto'),
 ('Matt Thomas', 1629744, 'Toronto'),
 ('Yuta Watanabe', 1629139, 'Toronto'),
 ('Quinn Cook', 1626188, 'Cleveland'),
 ('Damyean Dotson', 1628422, 'Cleveland'),
 ('Lamar Stevens', 1630205, 'Cleveland'),
 ('Brodric Thomas', 1630271, 'Cleveland'),
 ('Cristiano Felicio', 1626245, 'Chicago'),
 ('Chandler Hutchison', 1628990, 'Chicago'),
 ('Luke Kornet', 1628436, 'Chicago'),
 ('Adam Mokoka', 1629690, 'Chicago'),
 ('Sekou Doumbouya', 1629635, 'Detroit'),
 ('Deividas Sirvydis', 1629686, 'Detroit'),
 ('Terrance Ferguson', 1628390, 'Philadelphia'),
 ('Isaiah Joe', 1630198, 'Philadelphia'),
 ('Vincent Poirier', 1629738, 'Philadelphia'),
 ('Paul Reed', 1630194, 'Philadelphia'),
 ('Rayjon Tucker', 1629730, 'Philadelphia'),
 ('Ignas Brazdeikis', 1629649, 'New York'),
 ('Taj Gibson', 201959, 'New York'),
 ('Jared Harper', 1629607, 'New York'),
 ('Kevin Knox II', 1628995, 'New York'),
 ('Theo Pinson', 1629033, 'New York'),
 ('Marc Gasol', 201188, 'Los Angeles'),
 ('Frank Kaminsky', 1626163, 'Phoenix'),
 ("E'Twaun Moore", 202734, 'Phoenix'),
 ('Zach Collins', 1628380, 'Portland'),
 ('Jusuf Nurkic', 203994, 'Portland'),
 ('Bruno Fernando', 1628981, 'Atlanta'),
 ('Skylar Mays', 1630219, 'Atlanta'),
 ('Kostas Antetokounmpo', 1628961, 'Los Angeles'),
 ('Devontae Cacok', 1629719, 'Los Angeles'),
 ('Jabari Parker', 203953, 'Sacramento'),
 ('Ben Simmons', 1627732, 'Philadelphia'),
 ('Stephen Curry', 201939, 'Golden State'),
 ('Gorgui Dieng', 203476, 'Memphis'),
 ('John Konchar', 1629723, 'Memphis'),
 ('Jontay Porter', 1629007, 'Memphis'),
 ('Killian Tillie', 1629681, 'Memphis'),
 ('Xavier Tillman', 1630214, 'Memphis'),
 ('Trey Lyles', 1626168, 'San Antonio'),
 ('Patty Mills', 201988, 'San Antonio'),
 ('Dejounte Murray', 1627749, 'San Antonio'),
 ('Luka Samanic', 1629677, 'San Antonio'),
 ('Thanasis Antetokounmpo', 203648, 'Milwaukee'),
 ('Mamadi Diakite', 1629603, 'Milwaukee'),
 ('Axel Toupane', 1626253, 'Milwaukee'),
 ('Nate Darling', 1630268, 'Charlotte'),
 ('Nick Richards', 1630208, 'Charlotte'),
 ('Grant Riller', 1630203, 'Charlotte'),
 ('Lonnie Walker IV', 1629022, 'San Antonio'),
 ('Quinndary Weatherspoon', 1629683, 'San Antonio'),
 ('Kyle Guy', 1629657, 'Sacramento'),
 ('DaQuan Jeffries', 1629610, 'Sacramento'),
 ('Robert Woodard II', 1630218, 'Sacramento'),
 ('Tacko Fall', 1629605, 'Boston'),
 ('Semi Ojeleye', 1628400, 'Boston'),
 ('Jale

306 of my 1008 players recorded games which they did not play in. This is also something that is usefull to know because a very important part about being a good basketball player is the ability to be availble and play on any given night, although a strong majority of these players are reserves. 

5.Plusminus is an extremely important statistic in basketball to dereminte how effective someone is when they are on the court. This will be a great indicator that I will use to find out who the statistically most efficient basketball player is. I want to find the players with the best plus minus that scored over 20 points.

In [23]:
%%sql 
SELECT Player_name,
Team_city,
PLUS_MINUS
FROM game_details
WHERE PTS > 20 
ORDER BY PLUS_MINUS DESC
LIMIT 25;

 * mysql://admin:***@sqlproject.cumiuvaqlsmj.us-east-1.rds.amazonaws.com/sports_data
25 rows affected.


[('Tobias Harris', 'Philadelphia', 36),
 ('Zach LaVine', 'Chicago', 36),
 ('Giannis Antetokounmpo', 'Milwaukee', 34),
 ('Brandon Ingram', 'New Orleans', 33),
 ('LeBron James', 'Los Angeles', 33),
 ('Damian Lillard', 'Portland', 33),
 ('Kyrie Irving', 'Brooklyn', 32),
 ('Paul George', 'LA', 31),
 ('Giannis Antetokounmpo', 'Milwaukee', 31),
 ("De'Aaron Fox", 'Sacramento', 31),
 ('Kyrie Irving', 'Brooklyn', 31),
 ('Kevin Durant', 'Brooklyn', 31),
 ('Gordon Hayward', 'Charlotte', 30),
 ('Tobias Harris', 'Philadelphia', 30),
 ('CJ McCollum', 'Portland', 30),
 ('Jamal Murray', 'Denver', 30),
 ('Trae Young', 'Atlanta', 30),
 ('Josh Richardson', 'Dallas', 29),
 ('Luka Doncic', 'Dallas', 29),
 ('Luka Doncic', 'Dallas', 29),
 ('Seth Curry', 'Philadelphia', 29),
 ('Khris Middleton', 'Milwaukee', 28),
 ('Jrue Holiday', 'Milwaukee', 27),
 ('Kawhi Leonard', 'LA', 27),
 ('Andre Drummond', 'Cleveland', 27)]

Here is the list of the top 25 plus minus ratios from all the data of games I have collected. As you can see here this is also a very elite list of players who scored over 20 point and positvley affected their teams on both sides of the court. 

In order to help solve my business question of who the most efficient basketball player is I decided that I would make a data request to show all players performances based on PlusMinus. This woult help me understand how on any given night they impacted their team in either a positive or negative way. 

In [33]:
%%sql
SELECT DISTINCT player.Player_id,
player.Player_name,
game_details.Team_city,
game_details.PLUS_MINUS, 
CASE 
	WHEN PLUS_MINUS > 0 THEN 'good for team'
	WHEN PLUS_MINUS = 0 THEN 'nuetral'
	ELSE 'bad for team'
END AS overall_impact
FROM game_details
JOIN player
	ON player.Player_id = game_details.Player_id
WHERE PLUS_MINUS IS NOT NULL;

 * mysql://admin:***@sqlproject.cumiuvaqlsmj.us-east-1.rds.amazonaws.com/sports_data
3873 rows affected.


[(203200, 'Justin Holiday', 'Indiana', 8, 'good for team'),
 (1627734, 'Domantas Sabonis', 'Indiana', -2, 'bad for team'),
 (1626167, 'Myles Turner', 'Indiana', 6, 'good for team'),
 (1627747, 'Caris LeVert', 'Indiana', 1, 'good for team'),
 (1627763, 'Malcolm Brogdon', 'Indiana', 6, 'good for team'),
 (203926, 'Doug McDermott', 'Indiana', 2, 'good for team'),
 (204456, 'T.J. McConnell', 'Indiana', -1, 'bad for team'),
 (1628410, 'Edmond Sumner', 'Indiana', -3, 'bad for team'),
 (203087, 'Jeremy Lamb', 'Indiana', -5, 'bad for team'),
 (1629048, 'Goga Bitadze', 'Indiana', 3, 'good for team'),
 (202710, 'Jimmy Butler', 'Miami', -15, 'bad for team'),
 (203482, 'Kelly Olynyk', 'Miami', -17, 'bad for team'),
 (1628389, 'Bam Adebayo', 'Miami', 22, 'good for team'),
 (1629130, 'Duncan Robinson', 'Miami', 15, 'good for team'),
 (1629134, 'Kendrick Nunn', 'Miami', -9, 'bad for team'),
 (1629639, 'Tyler Herro', 'Miami', -5, 'bad for team'),
 (2738, 'Andre Iguodala', 'Miami', 17, 'good for team'),
 (2772, 'Trevor Ariza', 'Miami', -18, 'bad for team'),
 (1629676, 'Isaiah Roby', 'Oklahoma City', -4, 'bad for team'),
 (1629650, 'Moses Brown', 'Oklahoma City', 2, 'good for team'),
 (1629652, 'Luguentz Dort', 'Oklahoma City', -5, 'bad for team'),
 (1629660, 'Ty Jerome', 'Oklahoma City', 10, 'good for team'),
 (1629004, 'Svi Mykhailiuk', 'Oklahoma City', 2, 'good for team'),
 (1629026, 'Kenrich Williams', 'Oklahoma City', 9, 'good for team'),
 (1628382, 'Justin Jackson', 'Oklahoma City', -2, 'bad for team'),
 (1627863, 'Danuel House Jr.', 'Houston', -1, 'bad for team'),
 (1626174, 'Christian Wood', 'Houston', 9, 'good for team'),
 (203506, 'Victor Oladipo', 'Houston', 5, 'good for team'),
 (202322, 'John Wall', 'Houston', 6, 'good for team'),
 (1629645, 'Kevin Porter Jr.', 'Houston', -15, 'bad for team'),
 (1628425, 'Sterling Brown', 'Houston', -5, 'bad for team'),
 (1628391, 'D.J. Wilson', 'Houston', -11, 'bad for team'),
 (1627742, 'Brandon Ingram', 'New Orleans', 3, 'good for team'),
 (1629627, 'Zion Williamson', 'New Orleans', -3, 'bad for team'),
 (203500, 'Steven Adams', 'New Orleans', -5, 'bad for team'),
 (1629638, 'Nickeil Alexander-Walker', 'New Orleans', -3, 'bad for team'),
 (202339, 'Eric Bledsoe', 'New Orleans', -6, 'bad for team'),
 (1629740, 'Nicolo Melli', 'New Orleans', 8, 'good for team'),
 (1628404, 'Josh Hart', 'New Orleans', 12, 'good for team'),
 (1629637, 'Jaxson Hayes', 'New Orleans', 8, 'good for team'),
 (1629008, 'Michael Porter Jr.', 'Denver', 4, 'good for team'),
 (200794, 'Paul Millsap', 'Denver', 3, 'good for team'),
 (203999, 'Nikola Jokic', 'Denver', 3, 'good for team'),
 (203115, 'Will Barton', 'Denver', 3, 'good for team'),
 (1627750, 'Jamal Murray', 'Denver', -7, 'bad for team'),
 (1628408, 'PJ Dozier', 'Denver', -3, 'bad for team'),
 (203210, 'JaMychal Green', 'Denver', -7, 'bad for team'),
 (1628392, 'Isaiah Hartenstein', 'Denver', -10, 'bad for team'),
 (202329, 'Al-Farouq Aminu', 'Orlando', -8, 'bad for team'),
 (203932, 'Aaron Gordon', 'Orlando', -9, 'bad for team'),
 (202696, 'Nikola Vucevic', 'Orlando', -18, 'bad for team'),
 (203095, 'Evan Fournier', 'Orlando', -21, 'bad for team'),
 (203487, 'Michael Carter-Williams', 'Orlando', -16, 'bad for team'),
 (1628407, 'Dwayne Bacon', 'Orlando', -4, 'bad for team'),
 (1626184, 'Chasson Randle', 'Orlando', -2, 'bad for team'),
 (203920, 'Khem Birch', 'Orlando', 2, 'good for team'),
 (203516, 'James Ennis III', 'Orlando', 2, 'good for team'),
 (1629109, 'Gary Clark', 'Orlando', 1, 'good for team'),
 (1627759, 'Jaylen Brown', 'Boston', 23, 'good for team'),
 (1628369, 'Jayson Tatum', 'Boston', 7, 'good for team'),
 (1628464, 'Daniel Theis', 'Boston', 27, 'good for team'),
 (203935, 'Marcus Smart', 'Boston', 18, 'good for team'),
 (202689, 'Kemba Walker', 'Boston', 18, 'good for team'),
 (1629684, 'Grant Williams', 'Boston', -3, 'bad for team'),
 (1629057, 'Robert Williams III', 'Boston', -13, 'bad for team'),
 (1628400, 'Semi Ojeleye', 'Boston', 7, 'good for 

In this data request I created a view statement to help me see 1 home victory for every team that I have in my database. I wanted to see how many points they scored as well as every statistic associated with their shooting and passing to gain a better understanding of how they win basketball games.

In [42]:
%%sql
CREATE OR REPLACE VIEW hometeam_stats AS
	SELECT game_details.Team_city,
	games.PTS_home,
	games.FG_PCT_home,
	games.FG3_PCT_home,
	games.AST_home,
	games.REB_home
	FROM games
	JOIN game_details
		ON games.Game_id = game_details.Game_id
	WHERE Home_team_wins = 1
	GROUP BY Team_city;


SELECT *
FROM hometeam_stats;

 * mysql://admin:***@sqlproject.cumiuvaqlsmj.us-east-1.rds.amazonaws.com/sports_data
0 rows affected.
30 rows affected.


[('Golden State', 125, 0.457, 0.429, 24, 57),
 ('Brooklyn', 125, 0.457, 0.429, 24, 57),
 ('Milwaukee', 122, 0.475, 0.45, 23, 37),
 ('Boston', 122, 0.475, 0.45, 23, 37),
 ('Dallas', 106, 0.481, 0.367, 21, 48),
 ('Phoenix', 106, 0.481, 0.367, 21, 48),
 ('New Orleans', 111, 0.507, 0.432, 29, 35),
 ('Miami', 111, 0.507, 0.432, 29, 35),
 ('Los Angeles', 138, 0.56, 0.487, 33, 53),
 ('Charlotte', 121, 0.529, 0.467, 34, 50),
 ('Cleveland', 121, 0.529, 0.467, 34, 50),
 ('New York', 121, 0.489, 0.235, 28, 50),
 ('Indiana', 121, 0.489, 0.235, 28, 50),
 ('Orlando', 113, 0.477, 0.357, 23, 41),
 ('Washington', 113, 0.471, 0.286, 22, 47),
 ('Philadelphia', 113, 0.471, 0.286, 22, 47),
 ('Detroit', 111, 0.432, 0.268, 25, 48),
 ('Minnesota', 111, 0.432, 0.268, 25, 48),
 ('Toronto', 119, 0.47, 0.412, 36, 53),
 ('San Antonio', 119, 0.47, 0.412, 36, 53),
 ('Houston', 128, 0.461, 0.422, 27, 45),
 ('Portland', 128, 0.461, 0.422, 27, 45),
 ('Sacramento', 106, 0.453, 0.217, 20, 56),
 ('Atlanta', 128, 0.488, 0.476, 30, 45),
 ('Denver', 124, 0.529, 0.452, 31, 52),
 ('LA', 124, 0.534, 0.516, 32, 46),
 ('Memphis', 126, 0.538, 0.41, 27, 42),
 ('Chicago', 126, 0.494, 0.489, 27, 44),
 ('Utah', 106, 0.461, 0.357, 28, 46),
 ('Oklahoma City', 118, 0.523, 0.381, 31, 39)]

For my final data request I wanted to see all the players that scored more than the league average in pts. This will help me guauge which players are consistently high scorers and I can look down the list to find out who is an extreme outlier, this with a combination of plusminus will lead me to my answer. 

In [45]:
%%sql
WITH player_rank AS (
	SELECT DISTINCT player.Player_name,
		player.Player_id,
		game_details.Team_city,
		SUM(game_details.PTS) OVER (
			PARTITION BY player.Player_name
		) AS total_pts
	FROM player 
	JOIN game_details
		ON player.Player_id = game_details.Player_id 
)
SELECT *
FROM player_rank
WHERE total_pts > (
	SELECT AVG(total_pts)
	FROM player_rank
);

 * mysql://admin:***@sqlproject.cumiuvaqlsmj.us-east-1.rds.amazonaws.com/sports_data
172 rows affected.


[('Aaron Gordon', 203932, 'Orlando', Decimal('133')),
 ('Alec Burks', 202692, 'New York', Decimal('224')),
 ('Andrew Wiggins', 203952, 'Golden State', Decimal('286')),
 ('Bam Adebayo', 1628389, 'Miami', Decimal('219')),
 ('Ben Simmons', 1627732, 'Philadelphia', Decimal('199')),
 ('Bobby Portis', 1626171, 'Milwaukee', Decimal('141')),
 ('Bogdan Bogdanovic', 203992, 'Atlanta', Decimal('140')),
 ('Bojan Bogdanovic', 202711, 'Utah', Decimal('184')),
 ('Bradley Beal', 203078, 'Washington', Decimal('522')),
 ('Brandon Clarke', 1629634, 'Memphis', Decimal('178')),
 ('Brandon Ingram', 1627742, 'New Orleans', Decimal('372')),
 ('Brook Lopez', 201572, 'Milwaukee', Decimal('169')),
 ('Bryn Forbes', 1627854, 'Milwaukee', Decimal('119')),
 ('Buddy Hield', 1627741, 'Sacramento', Decimal('303')),
 ('Cameron Payne', 1626166, 'Phoenix', Decimal('124')),
 ('Caris LeVert', 1627747, 'Brooklyn', Decimal('156')),
 ('Caris LeVert', 1627747, 'Indiana', Decimal('156')),
 ('Carmelo Anthony', 2546, 'Portland', Decimal('227')),
 ('Chris Boucher', 1628449, 'Toronto', Decimal('199')),
 ('Chris Paul', 101108, 'Phoenix', Decimal('215')),
 ('Christian Wood', 1626174, 'Houston', Decimal('141')),
 ('CJ McCollum', 203468, 'Portland', Decimal('206')),
 ('Clint Capela', 203991, 'Atlanta', Decimal('129')),
 ('Coby White', 1629632, 'Chicago', Decimal('227')),
 ('Collin Sexton', 1629012, 'Cleveland', Decimal('377')),
 ('Cory Joseph', 202709, 'Sacramento', Decimal('122')),
 ('Damian Lillard', 203081, 'Portland', Decimal('479')),
 ('Daniel Theis', 1628464, 'Boston', Decimal('144')),
 ('Danilo Gallinari', 201568, 'Atlanta', Decimal('211')),
 ('Danny Green', 201980, 'Philadelphia', Decimal('130')),
 ('Darius Garland', 1629636, 'Cleveland', Decimal('241')),
 ('Davis Bertans', 202722, 'Washington', Decimal('156')),
 ("De'Aaron Fox", 1628368, 'Sacramento', Decimal('398')),
 ("De'Anthony Melton", 1629001, 'Memphis', Decimal('121')),
 ('Deandre Ayton', 1629028, 'Phoenix', Decimal('232')),
 ('Dejounte Murray', 1627749, 'San Antonio', Decimal('251')),
 ('DeMar DeRozan', 201942, 'San Antonio', Decimal('265')),
 ('Dennis Schroder', 203471, 'Los Angeles', Decimal('275')),
 ('Devin Booker', 1626164, 'Phoenix', Decimal('380')),
 ('Dillon Brooks', 1628415, 'Memphis', Decimal('276')),
 ('Domantas Sabonis', 1627734, 'Indiana', Decimal('316')),
 ('Donovan Mitchell', 1628378, 'Utah', Decimal('360')),
 ('Donte DiVincenzo', 1628978, 'Milwaukee', Decimal('191')),
 ('Dorian Finney-Smith', 1627827, 'Dallas', Decimal('119')),
 ('Doug McDermott', 203926, 'Indiana', Decimal('234')),
 ('Duncan Robinson', 1629130, 'Miami', Decimal('178')),
 ('Dwayne Bacon', 1628407, 'Orlando', Decimal('139')),
 ('Dwight Howard', 2730, 'Philadelphia', Decimal('134')),
 ('Enes Kanter', 202683, 'Portland', Decimal('190')),
 ('Eric Bledsoe', 202339, 'New Orleans', Decimal('154')),
 ('Evan Fournier', 203095, 'Orlando', Decimal('194')),
 ('Fred VanVleet', 1627832, 'Toronto', Decimal('146')),
 ('Furkan Korkmaz', 1627788, 'Philadelphia', Decimal('154')),
 ('Gary Trent Jr.', 1629018, 'Portland', Decimal('197')),
 ('Giannis Antetokounmpo', 203507, 'Milwaukee', Decimal('414')),
 ('Goran Dragic', 201609, 'Miami', Decimal('205')),
 ('Gordon Hayward', 202330, 'Charlotte', Decimal('210')),
 ('Harrison Barnes', 203084, 'Sacramento', Decimal('276')),
 ('Isaiah Roby', 1629676, 'Oklahoma City', Decimal('131')),
 ('Ivica Zubac', 1627826, 'LA', Decimal('148')),
 ('Ja Morant', 1629630, 'Memphis', Decimal('306')),
 ('Jae Crowder', 203109, 'Phoenix', Decimal('153')),
 ('Jakob Poeltl', 1627751, 'San Antonio', Decimal('123')),
 ('Jalen Brunson', 1628973, 'Dallas', Decimal('177')),
 ('Jamal Murray', 1627750, 'Denver', Decimal('320')),
 ('James Harden', 201935, 'Brooklyn', Decimal('371')),
 ('James Harden', 201935, 'Houston', Decimal('371')),
 ('Jarrett Allen', 1628386, 'Cleveland', Decimal('155')),
 ('Jarrett Allen', 1628386, 'Brooklyn', Decimal('155')),
 ('JaVale McGee', 201580, 'Cleveland', Decimal('119')),
 ('Jaylen Brown', 1627759, 'Bos